File: src/main/java/de/esymetric/jerusalem/ownDataRepresentation/geoData/Position.java
package de.esymetric.jerusalem.ownDataRepresentation.geoData;
public class Position {
        public double latitude;
        public double longitude;
        public double altitude;
        public int elapsedTimeSec;
        public byte nrOfTransitions;

        public boolean Equals(Object other)
        {
            if (!(other instanceof Position)) return false;
            return ((Position)other).latitude == latitude && ((Position)other).longitude == longitude;
        }

        public int GetHashCode()
        {
            return super.hashCode();
        }

        public boolean IsNull()
        {
            return latitude == 0.0 && longitude == 0.0;
        }

        public double distanceTo(Position p)
        {
            return GPSMath.CalculateDistance(p.latitude, p.longitude, latitude, longitude);
        }
    }

File: src/main/java/de/esymetric/jerusalem/ownDataRepresentation/geoData/Position.kt
package de.esymetric.jerusalem.ownDataRepresentation.geoData

class Position {
    @JvmField
    var latitude = 0.0
    @JvmField
    var longitude = 0.0
    var altitude = 0.0
    var elapsedTimeSec = 0
    var nrOfTransitions: Byte = 0
    fun Equals(other: Any?): Boolean {
        return if (other !is Position) false else other.latitude == latitude && other.longitude == longitude
    }

    fun GetHashCode(): Int {
        return super.hashCode()
    }

    fun IsNull(): Boolean {
        return latitude == 0.0 && longitude == 0.0
    }

    fun distanceTo(p: Position): Double {
        return GPSMath.calculateDistance(p.latitude, p.longitude, latitude, longitude)
    }
}